### ISSUE TITLE
Suggestion: Advanced tuner selection with groups

### ISSUE THREAD

**Issue by lars18th** [catalinii/minisatip#618]
Hi @catalinii ,

Based on the concept of the  `-S` parameter (master/slave tuner selection) I suggest to support a more advanced tuner selection. To understand the target scenario, please consider these examples:

- For the same sat position the tuner 0 is connected to HH, tuner 1 to HV, tuner 2 to LH and tuner 3 to LV. In this case, minisatip can't select at time the correct tuner.
- Another similar case: the tuner 0 is connected to HH and LH and tuner 1 to HV and LV. In this case the problem is similar.
- A completly different case: tuner 0 is connected to all bands, but tuner 1 is fixed to some frequency. And both are using the same sat position.

So the context can be defined as this: a group of tuners share the same sat position, but they can tune only part of the bands.

In this case we can use some file to describe the group. 
Examples:
`minisatip --group 0-3 group1.cfg`
```
0:9750-10600:*
1:9750-10600:*
2:10600-11700:*
3:10600-11700:*
```
`minisatip --group 0,1 group2.cfg`
```
0:*:V
1:*:H
```
`minisatip --group 0,1,3 group3.cfg`
```
0:*:*
1:10744:H
3:*:*
```

Then when selecting one adapter you only need to check if this adapter is inside a group and search for the correct one.

What you think?

**Comment by 9000h** [comment]
Hi,
is it really needed as today's UnicableII is that dirt cheap also for multi-feed, and the already implemented LNB sharing is questionable now.
CU
9000h

**Comment by lars18th** [comment]
> is it really needed as today's UnicableII is that dirt cheap also for multi-feed, and the already implemented LNB sharing is questionable now.

Yes, a pre-configured UnicableII LNBs group with multiple sat positions and fixed frequencies it's another scenario for this enhancement.

However, I'm not sure about the best _CFG_ format to describe the tuner group. Any suggestion?

**Comment by lars18th** [comment]
Hi,

Is not a full solution, but a partially solution: https://github.com/catalinii/minisatip/issues/619#issuecomment-545409503

Hi @9000h , perhaps you can help to finish the patch.
Regards.

**Comment by lars18th** [comment]
Hi,

I open this for discussion about the best mode to group tuners. This is my first proposal of command-line description...

Format: `-6 TUNER_NUMBER:FREQ1-FREQ2,POL,SRC;[...]`
Example: `-6 0:9750-10600,v,1;1:10601-11750,v,1;2:*,*,2;3:*,h,1`
Description of the example: Tuner0 is connected to SRC=1 and in the low band vertical polarization, the Tuner1 is in high band vertical polarization in SRC=1 too. But the Tuner2 is in SRC=2 with full bands/polarizations and Tuner3 is in the same SRC=1 but horizontal polarization of both high and low bands.

I feel this is simple to describe and sufficient powerful.
How you think?

**Comment by catalinii** [comment]
@lars18th , not sure if you saw but axe.c implements something similar: ability to connect 4 LNBs on different polarization and band to all inputs and then switch the input for the tuner 0-3 to the right input.

**Comment by lars18th** [comment]
> @lars18th , not sure if you saw but axe.c implements something similar: ability to connect 4 LNBs on different polarization and band to all inputs and then switch the input for the tuner 0-3 to the right input.

I'll, rewiew it.
However, I need some help (very busy at time) to almost implement the command-line part.
